libs/SmartSync/src/com/salesforce/androidsdk/smartsync/util/SyncOptions.java

```diff
@@ -106,13 +126,23 @@ public JSONObject asJSON() throws JSONException {
 		JSONObject options = new JSONObject();
         if (mergeMode != null) options.put(MERGEMODE, mergeMode.name());
 		if (fieldlist != null) options.put(FIELDLIST, new JSONArray(fieldlist));
+		if (createFieldlist != null) options.put(CREATE_FIELDLIST, new JSONArray(createFieldlist));
+		if (updateFieldlist != null) options.put(UPDATE_FIELDLIST, new JSONArray(updateFieldlist));
 		return options;
 	}
```

Comment:
The code making use of it is here: https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/1343/files#diff-4201649df1e9c61d455833a013ad7b4bR498
During sync up, there might be creates / updatse and deletes.
But a given record is either created or updated or deleted.
During create, the create list is used if available otherwise fieldlist is used.
During update, the update list is used if available otherwise fieldlist is used.